### PR TITLE
Fix #157: Change ?ds to a slug

### LINKED ISSUE #157: Use the experiment name in the URL, not the experiment ID
At the moment, the selected dataset is specified in the URL via the `ds` query parameter. For example: `?ds=9`.

Gregg and John mentioned that it would be great if we could show the experiment name in the URL instead. For example `?ds=pref-flip-whatever` or `?experiment=pref-flip-whatever`.

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

File: viewer/api/factories.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ class Meta:
 
 
 class MetricFactory(factory.django.DjangoModelFactory):
-    name = factory.Sequence(lambda n: 'Metric %s' % n)
+    name = factory.Sequence(lambda n: 'scalar_cats_a_purring_%s' % n)
```
Comment (openjck) [inline viewer/api/factories.py]: I'm confused. 🐈

**Comment by openjck** [comment]
I'm also looking into the error you were having locally. That's the `props.currentDataset is undefined` thing, right?

File: viewer/api/factories.py (hunk 1 above)
Comment (robhudson) [inline viewer/api/factories.py]: It was a purr-fect time to make the factory metrics look more like real metrics.

**Comment by robhudson** [comment]
> I'm also looking into the error you were having locally. That's the props.currentDataset is undefined thing, right?

Yep, that's the one.

**Comment by openjck** [comment]
It doesn't fix the console error, but *dataset-reducer.js* will also need to be updated to rename `selectedDatasetId` to `selectedDataset` to match *dataset-actions.js*.

File: viewer/api/factories.py (hunk 1 above)
Comment (openjck) [inline viewer/api/factories.py]: 😆 

So this is the sample data?

File: viewer/static/js/app/components/containers/dataset-control-container.js [hunk 2]
```diff
@@ -65,7 +65,7 @@ class DatasetControlContainer extends React.Component {
         handleCohortSwitch={this._handleCohortSwitch}
         isBtnDisabled={this.isBtnDisabled}
 
-        currentDatasetId={this.props.selectedDatasetId}
+        currentDataset={this.props.selectedDataset}
```
Comment (openjck) [inline viewer/static/js/app/components/containers/dataset-control-container.js]: This is where the issue stems from. This prop has the same name as the prop on line 77, but they are two different types of data. This is a name, the one on line 77 is an object containing all sorts of data about the dataset. *dataset-control.js* needs the latter, so I'm going to push a commit that renames this.

**Comment by robhudson** [comment]
Works for me locally now. Thanks @openjck ! r+

**Comment by openjck** [comment]
Code looks good, tests pass, behaves as expected. r+

**Pull request merged**